src/mainloop.rkt

```diff
@@ -33,10 +33,10 @@
 ;; extending, make sure this never gets too complicated to fit in your
 ;; head at once, because then global state is going to mess you up.
 
-(struct shellstate (table next-alts locs lowlocs patched) #:mutable)
+(struct shellstate (table next-alts locs locosts lowlocs patched) #:mutable)
```

Comment:
I was thinking we'd only have `locs` and `lowlocs` but we'd only change what gets assigned to `locs` and `lowlocs`, i.e., `locs` gets high error subexpressions and `lowlocs` gets high cost subexpressions. Yes, the names make no sense now!

Reply:
Wouldn't that mean that we are no longer gonna use lowlocs in main?

Reply:
I think that's right, but I'm not sure there's actually any value to using them right now.